test/models/white_ip_test.rb

```diff
@@ -38,6 +38,16 @@ def test_validates_ipv6_format
     assert white_ip.valid?
   end
 
+  def test_validates_include_empty_ipv4
+    white_ip = valid_white_ip
+
+    white_ip.update(ipv4: '')
```

Comment:
Not sure 100%, but as far as I remember `#update` method may bypass model validations.
I'd suggest to:
1. Create WhiteIp model (`white_ip = WhiteIp.new`)
2. assign `white_ip.ipv6`.
3. Assert no exception risen on saving `white_ip`
4. assert `white_ip.valid?`